frontend/app/src/components/wallet-bridge/Pairing.vue

```diff
@@ -0,0 +1,345 @@
+    }
+
+    // Connect using the URI
+    await kit.core.pairing.pair({ uri: pairUriVal });
+    addLog('Successfully paired with the Electron app, you can do transaction there.', 'success');
+
+    set(pairUri, '');
+  }
+  catch (error) {
+    addLog(`Connection failed: ${(error as Error).message}`, 'error');
+  }
+  finally {
+    set(isConnecting, false);
+  }
+}
+
+function getActiveSessions(): SessionTypes.Struct[] {
+  const kit = get(walletKit);
+  if (!kit)
+    return [];
+
+  const sessionsMap = kit.getActiveSessions() || {};
+  return Object.values(sessionsMap);
+}
+
+async function chainChanged(topic: string, chainId: string) {
+  const kit = get(walletKit);
+
+  const eipChainId = getEip155ChainId(chainId);
+
+  return kit?.emitSessionEvent({
+    chainId: eipChainId,
+    event: {
+      data: Number(chainId),
+      name: 'chainChanged',
+    },
+    topic,
+  });
+}
+
+async function accountsChanged(topic: string, chainId: string, address: string) {
+  const kit = get(walletKit);
+
+  const eipChainId = getEip155ChainId(chainId);
+
+  return kit?.emitSessionEvent({
+    chainId: eipChainId,
+    event: {
+      data: [address],
+      name: 'accountsChanged',
+    },
+    topic,
+  });
+}
+
+async function triggerTransaction(request: TransactionRequest): Promise<TransactionResponse> {
+  try {
+    const browserProvider = getBrowserProvider();
+    const signer = await browserProvider.getSigner();
+
+    return await signer.sendTransaction(request);
+  }
+  catch (error) {
+    console.error(error);
+    throw error;
+  }
+}
+
+async function onSessionProposal({ id, params }: WalletKitTypes.SessionProposal) {
+  const kit = get(walletKit);
+  const addressVal = get(address);
+  const chainId = get(connectedChainId);
+
+  if (!kit || !addressVal)
+    return;
+
+  const chainIds = [chainId];
+  try {
+    // ------- namespaces builder util ------------ //
+    const approvedNamespaces = buildApprovedNamespaces({
+      proposal: params,
+      supportedNamespaces: {
+        eip155: {
+          accounts: chainIds.map(item => getEip155ChainId(`${item}:${addressVal}`)),
+          chains: chainIds.map(item => getEip155ChainId(item)),
+          events: ['accountsChanged', 'chainChanged'],
+          methods: ['eth_sendTransaction', 'personal_sign'],
+        },
+      },
+    });
+    // ------- end namespaces builder util ------------ //
+
+    await kit.approveSession({
+      id,
+      namespaces: approvedNamespaces,
+    });
+
+    set(activeSessions, getActiveSessions());
+  }
+  catch (error) {
+    console.error(error);
+  }
+}
+
+async function onSessionRequest(event: WalletKitTypes.SessionRequest) {
+  const kit = get(walletKit);
+  assert(kit);
+
+  const { id, params, topic } = event;
+  try {
+    const request = params.request.params[0];
+    const result = await triggerTransaction(request);
+    const chainName = get(getChainName(getChainFromChainId(Number(result.chainId))));
```

Comment:
to many gets here, don't we have any single function to get the chain name from id?